### PR TITLE
Create bash remediation for file_permissions_unauthorized_world_writable

### PR DESCRIPTION
**Pull request by cyarbrough76**
#### Description:

- _Add remediation for world writable files_

### CODE REVIEW

**Comment by mildas** [comment]
**Changes identified:**
Rule file_permissions_unauthorized_world_writable:
 Bash remediation is newly added.

**Recommended tests to execute:**
 build_product ol8
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol8-ds.xml file_permissions_unauthorized_world_writable

File: linux_os/guide/system/permissions/files/file_permissions_unauthorized_world_writable/bash/shared.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# platform = multi_platform_rhel
+
+(cd / && find . -type f -perm -002 -not -path "./selinux/*" -not -path "./proc/*" -not -path "./sys/*" -not -path "./cgroup/*" -exec chmod -v o-w {} +)
```
Comment (redhatrises) [inline linux_os/guide/system/permissions/files/file_permissions_unauthorized_world_writable/bash/shared.sh:3]: This should be:

`find / -xdev -type f -perm -002 -exec chmod o-w {} \;`

**Comment by mildas** [comment]
**Changes identified:**
Rule file_permissions_unauthorized_world_writable:
 Bash remediation is newly added.

**Recommended tests to execute:**
 build_product ol8
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol8-ds.xml file_permissions_unauthorized_world_writable

File: linux_os/guide/system/permissions/files/file_permissions_unauthorized_world_writable/bash/shared.sh [hunk 2]
```diff
@@ -1,3 +1,3 @@
-# platform = multi_platform_rhel
+# platform = multi_platform_all
 
-(cd / && find . -type f -perm -002 -not -path "./selinux/*" -not -path "./proc/*" -not -path "./sys/*" -not -path "./cgroup/*" -exec chmod -v o-w {} +)
+(cd / && find / -xdev -type f -perm -002 -exec chmod o-w {} \;)
```
Comment (redhatrises) [inline linux_os/guide/system/permissions/files/file_permissions_unauthorized_world_writable/bash/shared.sh:3]: Seems like having `cd / && find /` is unnecessary. Shouldn't it just be `find / -xdev -type f -perm -002 -exec chmod o-w {} \;` ?

**Comment by mildas** [comment]
**Changes identified:**
Rule file_permissions_unauthorized_world_writable:
 Bash remediation is newly added.

**Recommended tests to execute:**
 build_product ol8
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol8-ds.xml file_permissions_unauthorized_world_writable

File: linux_os/guide/system/permissions/files/file_permissions_unauthorized_world_writable/bash/shared.sh (hunk 1 above)
Comment (matejak) [reply linux_os/guide/system/permissions/files/file_permissions_unauthorized_world_writable/bash/shared.sh]: I agree:

- `xdev` says "don't search directories that are mounted from different devices than the root's device", which is a simpler version of those `-not -path "..."`.
- find is position-independent when used with absolute paths.

File: linux_os/guide/system/permissions/files/file_permissions_unauthorized_world_writable/bash/shared.sh (hunk 2 above)
Comment (matejak) [reply linux_os/guide/system/permissions/files/file_permissions_unauthorized_world_writable/bash/shared.sh]: Yes, I agree.

**Comment by matejak** [comment]
I would recommend a squash merge if it passes the review - a one-liner is not worth three commits.

**Comment by matejak** [comment]
@redhatrises ping, I think that this one is ready for merging.

**Pull request merged**